File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Energy Monitor Bricklet

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim em As New BrickletEnergyMonitor(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get current Energy Data
        Dim voltage, current, energy, realPower, apparentPower, reactivePower,
            powerFactor, frequency As Integer

        em.GetEnergyData(voltage, current, energy, realPower, apparentPower, _
                         reactivePower, powerFactor, frequency)

        Console.WriteLine("Voltage: " + (voltage/100.0).ToString() + " V")
        Console.WriteLine("Current: " + (current/100.0).ToString() + " A")
        Console.WriteLine("Energy: " + (energy/100.0).ToString() + " Wh")
        Console.WriteLine("Real Power: " + (realPower/100.0).ToString() + " h")
        Console.WriteLine("Apparent Power: " + (apparentPower/100.0).ToString() + " VA")
        Console.WriteLine("Reactive Power: " + (reactivePower/100.0).ToString() + " var")
        Console.WriteLine("Power Factor: " + (powerFactor/1000.0).ToString())
        Console.WriteLine("Frequency: " + (frequency/100.0).ToString() + " Hz")

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/energy_monitor_bricklet"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Energy Monitor Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	em, _ := energy_monitor_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current Energy Data.
	voltage, current, energy, realPower, apparentPower, reactivePower, powerFactor, frequecy, _ := em.GetEnergyData()

	fmt.Printf("Voltage: %f V\n", float64(voltage)/100.0)
	fmt.Printf("Current: %f A\n", float64(current)/100.0)
	fmt.Printf("Energy: %f Wh\n", float64(energy)/100.0)
	fmt.Printf("Real Power: %f h\n", float64(realPower)/100.0)
	fmt.Printf("Apparent Power: %f VA\n", float64(apparentPower)/100.0)
	fmt.Printf("Reactive Power: %f VAR\n", float64(reactivePower)/100.0)
	fmt.Printf("Power Factor: %f\n", float64(powerFactor)/1000.0)
	fmt.Printf("Frequency: %f Hz\n", float64(frequecy)/100.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
